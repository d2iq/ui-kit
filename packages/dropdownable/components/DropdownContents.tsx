import React from "react";
import { useDetectClickOutside } from "react-detect-click-outside";

export interface DropdownContentsProps {
  children?: React.ReactNode | React.ReactNode[];
  isOpen?: boolean;
  onClose?: () => void;
}

const DropdownContents = ({ children, onClose }: DropdownContentsProps) => {
  const ref = useDetectClickOutside({
    onTriggered:
      onClose ??
      (() => {
        return null;
      })
  });

  return <div ref={ref}>{children}</div>;
};

export default DropdownContents;
