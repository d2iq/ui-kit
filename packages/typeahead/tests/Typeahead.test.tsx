import React from "react";
import { mount } from "enzyme";
import serializer from "@emotion/jest";
import toJson from "enzyme-to-json";
import { Typeahead, TextInput } from "../../index";
import DropdownContents from "../../dropdownable/components/DropdownContents";

expect.addSnapshotSerializer(serializer);

const items = [
  { label: "Exosphere", value: "Exosphere" },
  { label: "Thermosphere", value: "Thermosphere" },
  { label: "Stratosphere", value: "Stratosphere" }
];

describe("Typeahead", () => {
  it("renders", () => {
    const component = mount(
      <Typeahead
        items={items}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it("renders a menu with a max height", () => {
    const component = mount(
      <Typeahead
        items={items}
        menuMaxHeight={100}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it("opens the menu on focus", () => {
    const component = mount(
      <Typeahead
        items={[]}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );

    expect(component.find(DropdownContents).length).toBe(0);
    component.find("input").simulate("focus");
    expect(component.find(DropdownContents).prop("isOpen")).toBe(true);
  });

  it("opens the menu on click even if it's focused", () => {
    const component = mount(
      <Typeahead
        items={items}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );

    expect(component.find(DropdownContents).length).toBe(0);
    component.find("input").simulate("focus");
    component.find("input").simulate("keyDown", {
      key: "ArrowDown"
    });
    component.find("input").simulate("keyDown", {
      key: "Enter"
    });
    expect(component.find(DropdownContents).length).toBe(0);
    component.find("input").simulate("click");
    expect(component.find(DropdownContents).prop("isOpen")).toBe(true);
  });

  it("renders an empty state if no items are passed", () => {
    const component = mount(
      <Typeahead
        items={[]}
        menuEmptyState={<div id="emptyState">Empty</div>}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );

    component.find("input").simulate("focus");
    expect(component.find("#emptyState").exists()).toBe(true);
  });

  it("does not render an empty state if items are passed", () => {
    const component = mount(
      <Typeahead
        items={items}
        menuEmptyState={<div id="emptyState">Empty</div>}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );

    component.find("input").simulate("focus");
    expect(component.find("#emptyState").exists()).toBe(false);
  });

  it("renders whatever text input field that is passed", () => {
    const component = mount(
      <Typeahead items={items} textField={<input id="randomInput" />} />
    );

    expect(component.find("#randomInput").exists()).toBe(true);
  });

  it("calls onSelect prop with the selected values", () => {
    const onSelectFn = jest.fn();
    const component = mount(
      <Typeahead
        items={items}
        onSelect={onSelectFn}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );
    const input = component.find("input");

    input.simulate("focus");
    expect(onSelectFn).not.toHaveBeenCalled();
    input.simulate("keyDown", {
      key: "ArrowDown"
    });
    input.simulate("keyDown", {
      key: "Enter"
    });
    expect(onSelectFn).toHaveBeenCalledWith([items[0].value], items[0].value);
  });

  it("sets the input value to the selected item's value on select", () => {
    const onSelectFn = jest.fn();
    const component = mount(
      <Typeahead
        items={items}
        onSelect={onSelectFn}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );
    expect(component.find("input").prop("value")).toBe("");
    component.find("input").simulate("focus");
    expect(onSelectFn).not.toHaveBeenCalled();
    component.find("input").simulate("keyDown", {
      key: "ArrowDown"
    });
    component.find("input").simulate("keyDown", {
      key: "Enter"
    });
    expect(component.find("input").prop("value")).toBe(items[0].value);
  });

  it("does not set the input value to the selected item's value on select", () => {
    const onSelectFn = jest.fn();
    const component = mount(
      <Typeahead
        items={[
          { label: "K8sphere", value: "K8sphere", disabled: true },
          ...items
        ]}
        onSelect={onSelectFn}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );
    expect(component.find("input").prop("value")).toBe("");
    component.find("input").simulate("focus");
    expect(onSelectFn).not.toHaveBeenCalled();
    component.find("input").simulate("keyDown", {
      key: "ArrowDown"
    });
    component.find("input").simulate("keyDown", {
      key: "Enter"
    });
    expect(component.find("input").prop("value")).not.toBe(items[0].value);
  });

  it("does not hide the dropdown when selecting an item if multiSelect is true", () => {
    const onSelectFn = jest.fn();
    const component = mount(
      <Typeahead
        multiSelect={true}
        items={items}
        onSelect={onSelectFn}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );

    expect(component.find(DropdownContents).length).toBe(0);
    component.find("input").simulate("focus");
    expect(onSelectFn).not.toHaveBeenCalled();
    component.find("input").simulate("keyDown", {
      key: "ArrowDown"
    });
    component.find("input").simulate("keyDown", {
      key: "Enter"
    });
    expect(component.find(DropdownContents).prop("isOpen")).toBe(true);
  });

  it("hides the dropdown when selecting an item if keepOpenOnSelect is false", () => {
    const onSelectFn = jest.fn();
    const component = mount(
      <Typeahead
        multiSelect={true}
        keepOpenOnSelect={false}
        items={items}
        onSelect={onSelectFn}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );

    expect(component.find(DropdownContents).length).toBe(0);
    component.find("input").simulate("focus");
    expect(onSelectFn).not.toHaveBeenCalled();
    component.find("input").simulate("keyDown", {
      key: "ArrowDown"
    });
    component.find("input").simulate("keyDown", {
      key: "Enter"
    });
    expect(component.find(DropdownContents).length).toBe(0);
  });

  it("does not set the input value if multiSelect is true", () => {
    const onSelectFn = jest.fn();
    const component = mount(
      <Typeahead
        multiSelect={true}
        items={items}
        onSelect={onSelectFn}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );

    expect(component.find("input").prop("value")).toBe("");
    component.find("input").simulate("focus");
    expect(onSelectFn).not.toHaveBeenCalled();
    component.find("input").simulate("keyDown", {
      key: "ArrowDown"
    });
    component.find("input").simulate("keyDown", {
      key: "Enter"
    });
    expect(component.find("input").prop("value")).toBe("");
  });

  it("does not set the input value if resetInputOnSelect is true", () => {
    const onSelectFn = jest.fn();
    const component = mount(
      <Typeahead
        resetInputOnSelect={true}
        items={items}
        onSelect={onSelectFn}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
          />
        }
      />
    );

    expect(component.find("input").prop("value")).toBe("");
    component.find("input").simulate("focus");
    expect(onSelectFn).not.toHaveBeenCalled();
    component.find("input").simulate("keyDown", {
      key: "ArrowDown"
    });
    component.find("input").simulate("keyDown", {
      key: "Enter"
    });
    expect(component.find("input").prop("value")).toBe("");
  });

  it("calls the input's onClick prop", () => {
    const onClickFn = jest.fn();
    const component = mount(
      <Typeahead
        items={items}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
            onClick={onClickFn}
          />
        }
      />
    );

    expect(component.find("input").prop("value")).toBe("");
    expect(onClickFn).not.toHaveBeenCalled();
    component.find("input").simulate("click");
    expect(onClickFn).toHaveBeenCalled();
  });

  it("calls the input's onFocus prop", () => {
    const onFocusFn = jest.fn();
    const component = mount(
      <Typeahead
        items={items}
        textField={
          <TextInput
            id="standard.input"
            inputLabel="Standard"
            placeholder="Placeholder"
            onFocus={onFocusFn}
          />
        }
      />
    );

    expect(component.find("input").prop("value")).toBe("");
    expect(onFocusFn).not.toHaveBeenCalled();
    component.find("input").simulate("focus");
    expect(onFocusFn).toHaveBeenCalled();
  });
});
