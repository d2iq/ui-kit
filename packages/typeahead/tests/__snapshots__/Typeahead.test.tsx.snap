// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typeahead renders 1`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
}

.emotion-1 {
  -webkit-align-items: flex;
  -webkit-box-align: flex;
  -ms-flex-align: flex;
  align-items: flex;
  height: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
}

.emotion-1>div {
  width: auto;
}

.emotion-2 {
  background: none;
  border: 0;
  box-shadow: none;
  padding: 0;
  padding-left: 16px;
  padding-right: 16px;
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  height: 36px;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  background-color: var(--themeBgPrimary, #FFFFFF);
  border-color: var(--themeBorder, #DADDE2);
}

.emotion-2:focus {
  outline: 0;
}

.emotion-2::-webkit-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2::-moz-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2:-ms-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2::placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input {
  font-size: inherit;
  color: inherit;
}

.emotion-2 input::-webkit-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input::-moz-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input:-ms-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input::placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2:focus {
  background-color: rgba(125,88,255,0.05);
  border-color: var(--themeBrandPrimary, #7D58FF);
}

<Typeahead
  items={
    Array [
      Object {
        "label": "Exosphere",
        "value": "Exosphere",
      },
      Object {
        "label": "Thermosphere",
        "value": "Thermosphere",
      },
      Object {
        "label": "Stratosphere",
        "value": "Stratosphere",
      },
    ]
  }
  menuMaxHeight={300}
  textField={
    <TextInput
      appearance="standard"
      id="standard.input"
      inputLabel="Standard"
      placeholder="Placeholder"
      showInputLabel={true}
      type="text"
    />
  }
>
  <div
    data-cy="typeahead"
  >
    <Downshift
      defaultHighlightedIndex={null}
      defaultIsOpen={false}
      environment={[Window]}
      getA11yStatusMessage={[Function]}
      itemToString={[Function]}
      onChange={[Function]}
      onInputValueChange={[Function]}
      onOuterClick={[Function]}
      onSelect={[Function]}
      onStateChange={[Function]}
      onUserAction={[Function]}
      scrollIntoView={[Function]}
      selectedItemChanged={[Function]}
      stateReducer={[Function]}
      suppressRefError={false}
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        aria-owns={null}
        role="combobox"
      >
        <Dropdownable
          dropdown={
            <div
              data-cy="typeahead-dropdown"
            >
              .emotion-0 {
  margin-top: 4px;
}

<div
                className="emotion-0"
              >
                <Popover
                  aria-labelledby="downshift-0-label"
                  id="downshift-0-menu"
                  maxHeight={300}
                  menuRef={[Function]}
                  role="listbox"
                  width={null}
                >
                  <PopoverListItem
                    aria-selected={false}
                    id="downshift-0-item-0"
                    index={0}
                    isActive={false}
                    isSelected={null}
                    listLength={3}
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    role="option"
                  >
                    Exosphere
                  </PopoverListItem>
                  <PopoverListItem
                    aria-selected={false}
                    id="downshift-0-item-1"
                    index={1}
                    isActive={false}
                    isSelected={null}
                    listLength={3}
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    role="option"
                  >
                    Thermosphere
                  </PopoverListItem>
                  <PopoverListItem
                    aria-selected={false}
                    id="downshift-0-item-2"
                    index={2}
                    isActive={false}
                    isSelected={null}
                    listLength={3}
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    role="option"
                  >
                    Stratosphere
                  </PopoverListItem>
                </Popover>
              </div>
            </div>
          }
          isOpen={false}
        >
          <div>
            <TextInput
              appearance="standard"
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls={null}
              aria-labelledby="downshift-0-label"
              autoComplete="off"
              downshiftReset={[Function]}
              id="standard.input"
              inputLabel="Standard"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Placeholder"
              showInputLabel={true}
              type="text"
              value=""
            >
              <div
                data-cy="textInput textInput.standard"
              >
                <label
                  className="emotion-0"
                  data-cy="textInput-label"
                  htmlFor="standard.input"
                >
                  Standard
                </label>
                <FormFieldWrapper
                  id="standard.input"
                >
                  <div>
                    <div
                      className="emotion-1"
                    >
                      <input
                        aria-activedescendant={null}
                        aria-autocomplete="list"
                        aria-controls={null}
                        aria-describedby=""
                        aria-invalid={false}
                        aria-labelledby="downshift-0-label"
                        autoComplete="off"
                        className="emotion-2"
                        data-cy="textInput-input"
                        downshiftReset={[Function]}
                        id="standard.input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Placeholder"
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      data-cy="textInput-hintContent"
                    />
                  </div>
                </FormFieldWrapper>
              </div>
            </TextInput>
          </div>
        </Dropdownable>
      </div>
    </Downshift>
  </div>
</Typeahead>
`;

exports[`Typeahead renders a menu with a max height 1`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
}

.emotion-1 {
  -webkit-align-items: flex;
  -webkit-box-align: flex;
  -ms-flex-align: flex;
  align-items: flex;
  height: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
}

.emotion-1>div {
  width: auto;
}

.emotion-2 {
  background: none;
  border: 0;
  box-shadow: none;
  padding: 0;
  padding-left: 16px;
  padding-right: 16px;
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  height: 36px;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  background-color: var(--themeBgPrimary, #FFFFFF);
  border-color: var(--themeBorder, #DADDE2);
}

.emotion-2:focus {
  outline: 0;
}

.emotion-2::-webkit-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2::-moz-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2:-ms-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2::placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input {
  font-size: inherit;
  color: inherit;
}

.emotion-2 input::-webkit-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input::-moz-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input:-ms-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input::placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2:focus {
  background-color: rgba(125,88,255,0.05);
  border-color: var(--themeBrandPrimary, #7D58FF);
}

<Typeahead
  items={
    Array [
      Object {
        "label": "Exosphere",
        "value": "Exosphere",
      },
      Object {
        "label": "Thermosphere",
        "value": "Thermosphere",
      },
      Object {
        "label": "Stratosphere",
        "value": "Stratosphere",
      },
    ]
  }
  menuMaxHeight={100}
  textField={
    <TextInput
      appearance="standard"
      id="standard.input"
      inputLabel="Standard"
      placeholder="Placeholder"
      showInputLabel={true}
      type="text"
    />
  }
>
  <div
    data-cy="typeahead"
  >
    <Downshift
      defaultHighlightedIndex={null}
      defaultIsOpen={false}
      environment={[Window]}
      getA11yStatusMessage={[Function]}
      itemToString={[Function]}
      onChange={[Function]}
      onInputValueChange={[Function]}
      onOuterClick={[Function]}
      onSelect={[Function]}
      onStateChange={[Function]}
      onUserAction={[Function]}
      scrollIntoView={[Function]}
      selectedItemChanged={[Function]}
      stateReducer={[Function]}
      suppressRefError={false}
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-1-label"
        aria-owns={null}
        role="combobox"
      >
        <Dropdownable
          dropdown={
            <div
              data-cy="typeahead-dropdown"
            >
              .emotion-0 {
  margin-top: 4px;
}

<div
                className="emotion-0"
              >
                <Popover
                  aria-labelledby="downshift-1-label"
                  id="downshift-1-menu"
                  maxHeight={100}
                  menuRef={[Function]}
                  role="listbox"
                  width={null}
                >
                  <PopoverListItem
                    aria-selected={false}
                    id="downshift-1-item-0"
                    index={0}
                    isActive={false}
                    isSelected={null}
                    listLength={3}
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    role="option"
                  >
                    Exosphere
                  </PopoverListItem>
                  <PopoverListItem
                    aria-selected={false}
                    id="downshift-1-item-1"
                    index={1}
                    isActive={false}
                    isSelected={null}
                    listLength={3}
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    role="option"
                  >
                    Thermosphere
                  </PopoverListItem>
                  <PopoverListItem
                    aria-selected={false}
                    id="downshift-1-item-2"
                    index={2}
                    isActive={false}
                    isSelected={null}
                    listLength={3}
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    role="option"
                  >
                    Stratosphere
                  </PopoverListItem>
                </Popover>
              </div>
            </div>
          }
          isOpen={false}
        >
          <div>
            <TextInput
              appearance="standard"
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls={null}
              aria-labelledby="downshift-1-label"
              autoComplete="off"
              downshiftReset={[Function]}
              id="standard.input"
              inputLabel="Standard"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Placeholder"
              showInputLabel={true}
              type="text"
              value=""
            >
              <div
                data-cy="textInput textInput.standard"
              >
                <label
                  className="emotion-0"
                  data-cy="textInput-label"
                  htmlFor="standard.input"
                >
                  Standard
                </label>
                <FormFieldWrapper
                  id="standard.input"
                >
                  <div>
                    <div
                      className="emotion-1"
                    >
                      <input
                        aria-activedescendant={null}
                        aria-autocomplete="list"
                        aria-controls={null}
                        aria-describedby=""
                        aria-invalid={false}
                        aria-labelledby="downshift-1-label"
                        autoComplete="off"
                        className="emotion-2"
                        data-cy="textInput-input"
                        downshiftReset={[Function]}
                        id="standard.input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Placeholder"
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      data-cy="textInput-hintContent"
                    />
                  </div>
                </FormFieldWrapper>
              </div>
            </TextInput>
          </div>
        </Dropdownable>
      </div>
    </Downshift>
  </div>
</Typeahead>
`;
